tests/python/proton_tests/utils.py

```diff
@@ -72,6 +76,24 @@ def wait(self):
         self.event.wait(self.timeout)
 
 
+class ConnPropertiesServer(EchoServer):
+     def __init__(self, url, timeout):
+        EchoServer.__init__(self, url, timeout)
+        self.properties_received = False
+        self.offered_capabilities_received = False
+        self.desired_capabilities_received = False
+
+     def on_link_opening(self, event):
```

Comment:
this is fine; could also be on_connection_opening which might be clearer in intent, but this works also